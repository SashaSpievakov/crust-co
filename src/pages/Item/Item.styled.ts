/* eslint import/prefer-default-export: 0 */

import styled from 'styled-components';

const Error = styled.h2`
  color: ${({ theme }) => theme.colors.primary};

  @media screen and (max-width: 550px) {
    font-size: 24px;
    padding-bottom: 25px;
  }
`;

export { Error };
