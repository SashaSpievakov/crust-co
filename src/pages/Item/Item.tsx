import { useEffect } from 'react';
import { useParams } from 'react-router-dom';

import ErrorRequest from 'src/components/ErrorRequest/ErrorRequest';
import Loading from '../../components/UI/Loading/Loading';
import FullItemBlock from '../../components/FullItemBlock/FullItemBlock';
import itemAPI from '../../services/ItemService';

const Item = () => {
  const { id } = useParams();

  const { data, isLoading, isSuccess } = itemAPI.useFetchItemQuery(
    id as string,
    {
      refetchOnFocus: true,
    },
  );

  useEffect(() => {
    document.title = `Crust & Co. | Item`;
  }, []);

  return isLoading ? (
    <Loading />
  ) : isSuccess ? (
    <FullItemBlock item={data} />
  ) : (
    <ErrorRequest />
  );
};

export default Item;
