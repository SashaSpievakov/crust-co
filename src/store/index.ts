export * from './slices';
export type { AppDispatch, AppStore, RootState } from './store';
export { setupStore } from './store';
