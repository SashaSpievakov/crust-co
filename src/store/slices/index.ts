export type { CartSliceState } from './cart';
export { addItem, clearCart, removeItem, removeItems } from './cart';
export { selectIsLight, setTheme } from './theme';
