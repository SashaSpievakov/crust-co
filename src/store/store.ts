import {
  combineReducers,
  configureStore,
  PreloadedState,
} from '@reduxjs/toolkit';

import categorySlice from './slices/category/reducer/categoryReducer';
import sortSlice from './slices/sort/reducer/sortReducer';
import searchSlice from './slices/search/reducer/searchValueReducer';
import theme from './slices/theme/reducer/themeReducer';
import cart from './slices/cart/reducer/cartReducer';
import itemAPI from '../services/ItemService';
import pizzasAPI from '../services/PizzasService';

const rootReducer = combineReducers({
  activeCategory: categorySlice,
  activeSort: sortSlice,
  searchValue: searchSlice,
  theme,
  cart,
  [itemAPI.reducerPath]: itemAPI.reducer,
  [pizzasAPI.reducerPath]: pizzasAPI.reducer,
});

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware()
        .concat(itemAPI.middleware)
        .concat(pizzasAPI.middleware),
    preloadedState,
  });
};

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;
export type AppDispatch = AppStore['dispatch'];
