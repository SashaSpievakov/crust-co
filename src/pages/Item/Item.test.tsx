import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { rest } from 'msw';

import Item from './Item';
import renderWithProviders from '../../tests/helpers/renderWithProviders';
import server from '../../tests/mocks/api/server';
import { setupStore } from '../../store/store';
import itemAPI from '../../services/ItemService';

describe('Item Tests', () => {
  beforeAll(() => {
    server.listen();
  });

  afterAll(() => {
    server.close();
  });

  afterEach(() => {
    server.resetHandlers();
    setupStore().dispatch(itemAPI.util.resetApiState());
  });

  test('renders the Item page', async () => {
    renderWithProviders(null, true, '/item/9');
    const itemTitle = await screen.findByText(/chicken curry/i); // eslint-disable-line @typescript-eslint/no-unused-vars

    const loadedItem = screen.getByTestId('itemPage');
    expect(loadedItem).toMatchSnapshot();
  });

  test('renders the Item page with a server requesr error', async () => {
    server.use(rest.get('*', (_req, res, ctx) => res.once(ctx.status(500))));
    renderWithProviders(<Item />);

    expect(
      await screen.findByText(/failed to get data from the server/i),
    ).toBeInTheDocument();
  });
});
