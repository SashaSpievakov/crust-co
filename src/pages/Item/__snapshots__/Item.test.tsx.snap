// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Tests renders the Item page 1`] = `
<article
  data-testid="itemPage"
>
  <article
    class="sc-bcXHqe sc-gswNZR bZXhLi edAeMk"
  >
    <img
      alt="pizza"
      class="sc-dkrFOg fXUcwl"
      height="450"
      src="../assets/img/pizza9.png"
      width="450"
    />
    <h2
      class="sc-hLBbgP fiXyOf"
    >
      Chicken Curry
    </h2>
    <p
      class="sc-eDvSVe jzLwQf"
    >
      Lorem, ipsum dolor sit amet consectetur adipisicing elit. Aliquid nisi aspernatur debitis quod suscipit, vitae placeat natus ipsum inventore esse rerum animi facere numquam saepe vero mollitia quibusdam at voluptates est commodi laudantium? Iste inventore quibusdam cupiditate nemo sint iusto minus nihil culpa! Sequi architecto nesciunt explicabo mollitia. Laboriosam, odio.
    </p>
    <section
      class="sc-gKPRtg eVocHQ"
    >
      <ul>
        <li
          class="sc-iBYQkv uIUYd"
        >
          traditional
        </li>
        <li
          class="sc-iBYQkv gouikj"
        >
          thin
        </li>
      </ul>
      <ul>
        <li
          class="sc-iBYQkv uIUYd"
        >
          12
           inch
        </li>
        <li
          class="sc-iBYQkv gouikj"
        >
          14
           inch
        </li>
        <li
          class="sc-iBYQkv gouikj"
        >
          16
           inch
        </li>
      </ul>
    </section>
    <h3
      class="sc-jSUZER glJRmG"
    >
      14
      $
    </h3>
    <button
      class="sc-ftTHYK sc-ipEyDJ sc-eDWCr KXpKQ ldgyoh lkfFKT"
    >
      <svg
        class="sc-bjfHbI lhpcBk"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0a1 1 0 0 1 1 1v6h6a1 1 0 1 1 0 2H9v6a1 1 0 1 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 1-1z"
        />
      </svg>
      <span>
        Add
      </span>
    </button>
    <a
      class="sc-ftTHYK sc-pyfCe KXpKQ gZyrww"
      href="/"
    >
      <span>
        Go Back
      </span>
    </a>
  </article>
</article>
`;

exports[`Item Tests renders the Item page with a server requesr error 1`] = `
<article
  data-testid="itemPage"
>
  <h2
    class="sc-dmctIk caGNgQ"
  >
    Error: failed request, try again
  </h2>
</article>
`;
